### PR TITLE
Add progress line

### PR DESCRIPTION
**Pull request by coreyfloyd**
and tint it blue - cuz that how we roll now.

### CODE REVIEW

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m [hunk 1]
```diff
@@ -433,13 +508,21 @@ - (void)saveWebViewScrollOffset {
...
-                                 .then(^(MWKArticle* article) {
+    [self showProgressViewAnimated:YES];
+    self.articleFetcherPromise = [self.articleFetcher fetchArticleForPageTitle:self.articleTitle progress:^(CGFloat progress) {
+        dispatchOnMainQueue(^{
+            [self updateProgress:[self totalProgressWithArticleFetcherProgress:progress] animated:YES];
+        });
+    }].then(^(MWKArticle* article) {
         @strongify(self);
         [self saveWebViewScrollOffset];
+        dispatchOnMainQueue(^{
```
Comment (bgerstle) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: we're already on the main queue, is dispatch really necessary?

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m [hunk 2]
```diff
@@ -433,13 +508,21 @@ - (void)saveWebViewScrollOffset {
...
+            [self updateProgress:0.85 animated:YES];
```
Comment (bgerstle) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: why 0.85?

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m [hunk 3]
```diff
@@ -366,6 +379,66 @@ - (UIBarButtonItem*)languagesToolbarItem {
...
+- (void)_hideProgressView {
+    self.progressView.alpha = 0.0;
+}
+
+- (void)updateProgress:(CGFloat)progress animated:(BOOL)animated {
+    [self.progressView setProgress:progress animated:animated];
+}
+
+- (CGFloat)totalProgressWithArticleFetcherProgress:(CGFloat)progress {
+    return 0.75 * progress;
```
Comment (bgerstle) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: FYI we can use the `NSProgress` API to mange this sort of thing in the future

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m [hunk 4]
```diff
@@ -251,6 +252,18 @@ - (nullable WMFShareOptionsController*)shareOptionsController {
     return _shareOptionsController;
 }
 
+- (UIProgressView*)progressView {
```
Comment (bgerstle) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: :+1: for going w/ out-of-the-box classes

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m (hunk 1 above)
Comment (coreyfloyd) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: yeah - they are needed to get the UI updating progressively - for some reason just doing git in line tends to delay getting updates until the very end.

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m (hunk 3 above)
Comment (coreyfloyd) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: Yeah - so far thats not been a great API - don't see many people really using that.

File: Wikipedia/UI-V5/WMFArticleContainerViewController.m (hunk 2 above)
Comment (coreyfloyd) [inline Wikipedia/UI-V5/WMFArticleContainerViewController.m]: @bgerstle pushed an update for this.

**Comment by montehurd** [comment]
Seems solid!

**Comment by bgerstle** [comment]
@coreyfloyd i removed all my commits, re-cherry-picking your new ones. i'm going to try separately to figure out the build timeout issue in travis, but don't want to block this from getting in master since it seems OK

**Comment by bgerstle** [comment]
tests passed locally on my machine, so merging

**Pull request merged**